crawler/file_processing.py

```diff
@@ -663,60 +717,394 @@ def format_and_filter_rows(self, csvreader: DictReader) -> Any:
+        """
+        # first extract any recognised columns
+        modified_row = self.filtered_row(row, line_number)
+
+        # add the centre name as source
+        modified_row[FIELD_SOURCE] = self.centre_config["name"]
 
         barcode_regex = self.centre_config["barcode_regex"]
         barcode_field = self.centre_config["barcode_field"]
 
-        for row in csvreader:
-            # only process rows that contain something in the cells
-            if self.row_valid_structure(row, line_number):
-                row = self.filtered_row(row, line_number)
-                row[FIELD_SOURCE] = self.centre_config["name"]
-                row[FIELD_PLATE_BARCODE] = None  # type: ignore
+        # extract the barcode and well coordinate
+        modified_row[FIELD_PLATE_BARCODE] = None  # type: ignore
+        if modified_row[barcode_field] and barcode_regex:
+            modified_row[FIELD_PLATE_BARCODE], modified_row[FIELD_COORDINATE] = self.extract_plate_barcode_and_coordinate(
+                modified_row, line_number, barcode_field, barcode_regex
+            )
 
-                if row[barcode_field] and barcode_regex:
-                    row[FIELD_PLATE_BARCODE], row[FIELD_COORDINATE] = self.extract_fields(
-                        row, line_number, barcode_field, barcode_regex
-                    )
+        # if we failed to extract the plate barcode row is invalid
+        if not modified_row[FIELD_PLATE_BARCODE]:
+            return None
 
-                if row[FIELD_PLATE_BARCODE]:
-                    row[FIELD_LINE_NUMBER] = line_number  # type: ignore
-                    row[FIELD_FILE_NAME] = self.file_name
-                    row[FIELD_FILE_NAME_DATE] = self.file_name_date()
-                    row[FIELD_CREATED_AT] = import_timestamp
-                    row[FIELD_UPDATED_AT] = import_timestamp
+        # check if the result value matches to expected
+        if not self.row_result_value_valid(modified_row, line_number):
+            return None
 
-                    row_signature = self.get_row_signature(row)
+        # check if the ct channel target values match to expected (if present)
+        if not self.row_channel_target_values_valid(modified_row, line_number):
+            return None
 
-                    if row_signature in seen_rows:
-                        logger.debug(f"Skipping {row_signature}: duplicate")
-                        self.logging_collection.add_error(
-                            "TYPE 5",
-                            f"Duplicated, line: {line_number}, root_sample_id: {row[FIELD_ROOT_SAMPLE_ID]}",
-                        )
-                        continue
-                    seen_rows.add(row_signature)
-                    augmented_data.append(row)
-                else:
-                    missing_data_count += 1
+        # check if the ct channel result values match to expected (if present)
```

Comment:
remove comment above and equivalent ones?  I think the method names are v self-explanatory

Reply:
Removed